src/win/core.c

```diff
@@ -442,3 +443,32 @@ int uv__socket_sockopt(uv_handle_t* handle, int optname, int* value) {
 
   return 0;
 }
+
+
+int uv_homedir(char* home, size_t* len) {
+  wchar_t* path;
+  HRESULT hr;
+  int r;
+
+  if (home == NULL)
+    return -EINVAL;
+
+  hr = SHGetKnownFolderPath(FOLDERID_Profile,
+                            0,
+                            NULL,
+                            &path);
+
+  if (hr != S_OK)
+    return uv_translate_sys_error(WSAGetLastError());
+
+  r = wcstombs(home, path, len);
```

Comment:
* use `len - 1` because otherwise adding the null terminator (couple lines below) could cause a buffer overrun.
* I'd prefer to use `WideCharToMultiByte` instead of `wcstombs` like we do everywhere in libuv.